common/assertion-functions/src/main/kotlin/com/testerum/common_assertion_functions/functions/builtin/ArrayFunctions.kt

```diff
@@ -0,0 +1,108 @@
+    fun isArray(actualNode: JsonNode) {
+        if (actualNode is ArrayNode) {
+            return
+        }
+
+        throw AssertionFailedException("expected an Array, but got ${actualNode.toString()}")
+    }
+    
+    @AssertionFunction
+    fun isArrayOrNull(actualNode: JsonNode) {
+        if (actualNode is NullNode) {
+            return
+        }
+
+        if (actualNode is ArrayNode) {
+            return
+        }
+
+        throw AssertionFailedException("expected an Array, but got ${actualNode.toString()}")
+    }
+
+    @AssertionFunction
+    fun isNonEmptyArray(actualNode: JsonNode) {
+        if (actualNode is ArrayNode) {
+            val arrayNode = actualNode.get(0)
+            if (arrayNode != null) {
+                val actualSize = arrayNode.size()
+                if (actualSize > 0) {
+                    return
+                } else {
+                    throw AssertionFailedException("expected an non empty Array, but got an Array with ${actualSize} ${getElementsWord(actualSize)}: ${arrayNode.toString()}")
+                }
+            }
+        }
+
+        throw AssertionFailedException("expected an non empty Array, but got ${actualNode.toString()}")
+    }
+
+    @AssertionFunction
+    fun isArrayWithSize(actualNode: JsonNode, expectedValue: BigDecimal) {
+        if (actualNode is ArrayNode) {
+            val arrayNode = actualNode.get(0)
+            if (arrayNode != null) {
+                val expectedSize = expectedValue.intValueExact()
+                val actualSize = arrayNode.size()
+                if (actualSize == expectedSize) {
+                    return
+                } else {
+                    throw AssertionFailedException("expected an Array with $expectedSize ${getElementsWord(expectedSize)}, but got an Array with $actualSize ${getElementsWord(actualSize)}: ${arrayNode.toString()}")
+                }
+            }
+        }
+
+        throw AssertionFailedException("expected an Array, but got ${actualNode.toString()}")
+    }
+
+    @AssertionFunction
+    fun isArrayWithMoreElementsThan(actualNode: JsonNode, expectedValue: BigDecimal) {
+        if (actualNode is ArrayNode) {
+            val arrayNode = actualNode.get(0)
+            if (arrayNode != null) {
+                val expectedSize = expectedValue.intValueExact()
+                val actualSize = arrayNode.size()
+                if(actualSize > expectedSize) {
+                    return
+                } else {
+                    throw AssertionFailedException("expected an Array with more than $expectedSize ${getElementsWord(expectedSize)}, but got an Array with $actualSize ${getElementsWord(actualSize)}: ${arrayNode.toString()}")
+                }
+            }
+        }
+
+        throw AssertionFailedException("expected an Array, but got ${actualNode.toString()}")
+    }
+
+    @AssertionFunction
+    fun isArrayWithLessElementsThan(actualNode: JsonNode, expectedValue: BigDecimal) {
```

Comment:
``isArrayWithSizeLessThan``?

Reply:
if you have a better idea please tell me